diff -urN src.orig/ros/rosbash/scripts/rosrun src/ros/rosbash/scripts/rosrun
--- src.orig/ros/rosbash/scripts/rosrun	2015-10-13 11:51:24.000000000 -0700
+++ src/ros/rosbash/scripts/rosrun	2015-11-19 09:07:06.432666100 -0800
@@ -68,7 +68,7 @@
     _perm="/111"
   fi
   debug "Searching for $2 with permissions $_perm"
-  exepathlist="`find -L "${catkin_package_libexec_dirs[@]}" "$pkgdir" -name "$2" -type f  -perm "$_perm" ! -regex ".*$pkgdir\/build\/.*" | uniq`"
+  exepathlist="`find -L "${catkin_package_libexec_dirs[@]}" "$pkgdir" \( -name "$2" -o -name "$2.exe" \) -type f  -perm "$_perm" ! -regex ".*$pkgdir\/build\/.*" | uniq`"
   IFS=$'\n'
   exepathlist=($exepathlist)
   unset IFS
