diff -urN urdfdom.orig/urdf_parser/include/urdf_parser/exportdecl.h urdfdom/urdf_parser/include/urdf_parser/exportdecl.h
--- urdfdom.orig/urdf_parser/include/urdf_parser/exportdecl.h	2016-05-24 17:29:06.600369600 -0700
+++ urdfdom/urdf_parser/include/urdf_parser/exportdecl.h	2016-05-24 11:31:04.413365700 -0700
@@ -1,4 +1,4 @@
-/*********************************************************************
+ /*********************************************************************
 * Software License Agreement (BSD License)
 *
 *  Copyright (c) 2008, Willow Garage, Inc.
@@ -74,7 +74,7 @@
 # else
 // Depending on whether one is building or using the
 // library define DLLAPI to import or export.
-#  ifdef URDFDOM_EXPORTS
+#  if defined(urdfdom_model_EXPORTS) || defined(urdfdom_world_EXPORTS)
 #   define URDFDOM_DLLAPI URDFDOM_DLLEXPORT
 #  else
 #   define URDFDOM_DLLAPI URDFDOM_DLLIMPORT
