diff -urN src.orig/dynamic_reconfigure/cmake/dynamic_reconfigure-macros.cmake src.fix/dynamic_reconfigure/cmake/dynamic_reconfigure-macros.cmake
--- src.orig/dynamic_reconfigure/cmake/dynamic_reconfigure-macros.cmake	2015-04-22 12:24:37.000000000 -0700
+++ src.fix/dynamic_reconfigure/cmake/dynamic_reconfigure-macros.cmake	2015-11-19 09:12:17.029722700 -0800
@@ -38,12 +38,21 @@
 
     #file(WRITE ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_PYTHON_DESTINATION}/cfg/__init__.py)
 
+	if(CYGWIN AND "${CMAKE_SIZEOF_VOID_P}" EQUAL "4")
     add_custom_command(OUTPUT
       ${_output_cpp} ${_output_dox} ${_output_usage} ${_output_py} ${_output_wikidoc}
-      COMMAND ${_cmd} 
+      COMMAND ${_cmd} || echo Ignoring dynamic_reconfigure failure...
       DEPENDS ${_input} ${gencfg_build_files}
       COMMENT "Generating dynamic reconfigure files from ${_cfg}: ${_output_cpp} ${_output_py}"
     )
+	else()
+    add_custom_command(OUTPUT
+      ${_output_cpp} ${_output_dox} ${_output_usage} ${_output_py} ${_output_wikidoc}
+      COMMAND ${_cmd}
+      DEPENDS ${_input} ${gencfg_build_files}
+      COMMENT "Generating dynamic reconfigure files from ${_cfg}: ${_output_cpp} ${_output_py}"
+    )
+	endif()
 
     list(APPEND ${PROJECT_NAME}_generated
       ${_output_cpp} ${_output_py}
diff -urN src.orig/dynamic_reconfigure/test/CMakeLists.txt src.fix/dynamic_reconfigure/test/CMakeLists.txt
--- src.orig/dynamic_reconfigure/test/CMakeLists.txt	2015-04-22 12:24:37.000000000 -0700
+++ src.fix/dynamic_reconfigure/test/CMakeLists.txt	2015-11-19 11:22:04.364350800 -0800
@@ -4,4 +4,6 @@
 
 add_dependencies(tests dynamic_reconfigure-ref_server)
 
+if(CATKIN_ENABLE_TESTING)
 add_rostest(test_python_simple_client.launch)
+endif()
\ No newline at end of file
