diff -urN src.orig/image_common/camera_calibration_parsers/CMakeLists.txt src/image_common/camera_calibration_parsers/CMakeLists.txt
--- src.orig/image_common/camera_calibration_parsers/CMakeLists.txt	2015-07-28 13:52:37.000000000 -0700
+++ src/image_common/camera_calibration_parsers/CMakeLists.txt	2015-09-03 14:00:30.148560900 -0700
@@ -15,7 +15,7 @@
 
 find_package(PkgConfig)
 
-if (APPLE)
+if (APPLE OR CYGWIN)
   #find yaml cpp for mac os
   find_library(YAML_CPP_LIBRARIES NAMES yaml-cpp)
   find_path(YAML_CPP_H_INCLUDE_DIR yaml-cpp/yaml.h )
