diff -ur src.orig/robot_model/collada_parser/CMakeLists.txt src/robot_model/collada_parser/CMakeLists.txt
--- src.orig/robot_model/collada_parser/CMakeLists.txt	2014-11-30 11:22:09.000000000 -0800
+++ src/robot_model/collada_parser/CMakeLists.txt	2015-02-27 19:45:59.410746000 -0800
@@ -18,7 +18,7 @@
 
 set(CMAKE_MODULE_PATH  ${PROJECT_SOURCE_DIR}/cmake-extensions/)
 find_package(PkgConfig)
-find_package(COLLADA_DOM 2.3 COMPONENTS 1.5)
+find_package(COLLADA_DOM 2.4 COMPONENTS 1.5)
 if(COLLADA_DOM_FOUND)
   include_directories(${COLLADA_DOM_INCLUDE_DIRS})
   link_directories(${COLLADA_DOM_LIBRARY_DIRS})
diff -ur src.orig/robot_model/collada_urdf/CMakeLists.txt src/robot_model/collada_urdf/CMakeLists.txt
--- src.orig/robot_model/collada_urdf/CMakeLists.txt	2014-11-30 11:22:21.000000000 -0800
+++ src/robot_model/collada_urdf/CMakeLists.txt	2015-02-28 16:23:41.573702200 -0800
@@ -4,6 +4,7 @@
 find_package(catkin REQUIRED COMPONENTS angles collada_parser resource_retriever urdf geometric_shapes tf cmake_modules)
 
 find_package(TinyXML REQUIRED)
+find_package(octomap REQUIRED)
 
 catkin_package(
   LIBRARIES ${PROJECT_NAME}
@@ -13,7 +14,7 @@
 include_directories(include)
 
 find_package(assimp QUIET)
-if ( NOT ASSIMP_FOUND )
+if ( NOT ASSIMP_FOUND AND NOT CYGWIN )
   find_package(Assimp QUIET)
   if ( NOT ASSIMP_FOUND )
     find_package(PkgConfig REQUIRED)
diff -ur src.orig/robot_model/collada_urdf/src/collada_to_urdf.cpp src/robot_model/collada_urdf/src/collada_to_urdf.cpp
--- src.orig/robot_model/collada_urdf/src/collada_to_urdf.cpp	2014-11-30 11:22:21.000000000 -0800
+++ src/robot_model/collada_urdf/src/collada_to_urdf.cpp	2015-02-28 12:54:05.735768800 -0800
@@ -13,12 +13,12 @@
 #include <assimp/postprocess.h>
 #include <assimp/scene.h>
 #else
-#include <assimp.hpp>
+#include <assimp/assimp.hpp>
 #if defined(ASSIMP_EXPORT_API)
 #include <assimp/export.hpp>
 #endif
-#include <aiScene.h>
-#include <aiPostProcess.h>
+#include <assimp/aiScene.h>
+#include <assimp/aiPostProcess.h>
 #endif
 
 #include <iostream>
diff -ur src.orig/robot_model/collada_urdf/src/collada_urdf.cpp src/robot_model/collada_urdf/src/collada_urdf.cpp
--- src.orig/robot_model/collada_urdf/src/collada_urdf.cpp	2014-11-30 11:22:21.000000000 -0800
+++ src/robot_model/collada_urdf/src/collada_urdf.cpp	2015-02-28 12:53:00.806276500 -0800
@@ -71,12 +71,12 @@
 #include <assimp/Importer.hpp>
 #include <assimp/postprocess.h>
 #else
-#include <assimp.hpp>
-#include <aiScene.h>
-#include <aiPostProcess.h>
-#include <DefaultLogger.h>
-#include <IOStream.h>
-#include <IOSystem.h>
+#include <assimp/assimp.hpp>
+#include <assimp/aiScene.h>
+#include <assimp/aiPostProcess.h>
+#include <assimp/DefaultLogger.h>
+#include <assimp/IOStream.h>
+#include <assimp/IOSystem.h>
 #endif
 
 #include <geometric_shapes/shapes.h>
Only in src/robot_model: robot_model.patch
diff -ur src.orig/robot_model/urdf/CMakeLists.txt src/robot_model/urdf/CMakeLists.txt
--- src.orig/robot_model/urdf/CMakeLists.txt	2014-11-30 11:22:12.000000000 -0800
+++ src/robot_model/urdf/CMakeLists.txt	2015-02-26 20:27:40.142992100 -0800
@@ -48,7 +48,7 @@
 install(TARGETS ${PROJECT_NAME} 
   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
+  RUNTIME DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION})
 
 install(DIRECTORY include/${PROJECT_NAME}/
   DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
