diff -urN src.orig/rqt_common_plugins/rqt_bag_plugins/src/rqt_bag_plugins/image_helper.py src/rqt_common_plugins/rqt_bag_plugins/src/rqt_bag_plugins/image_helper.py
--- src.orig/rqt_common_plugins/rqt_bag_plugins/src/rqt_bag_plugins/image_helper.py	2015-07-24 14:17:18.000000000 -0700
+++ src/rqt_common_plugins/rqt_bag_plugins/src/rqt_bag_plugins/image_helper.py	2015-11-18 20:07:24.889049200 -0800
@@ -36,7 +36,7 @@
 
 from PIL import Image
 from PIL import ImageOps
-import cairo
+import cairocffi as cairo
 
 
 def imgmsg_to_pil(img_msg, rgba=True):
