diff -urN src.orig/rviz/CMakeLists.txt src.fix/rviz/CMakeLists.txt
--- src.orig/rviz/CMakeLists.txt	2015-10-13 10:52:04.000000000 -0700
+++ src.fix/rviz/CMakeLists.txt	2015-11-19 09:12:17.905810300 -0800
@@ -17,12 +17,16 @@
   thread
 )
 
-find_package(PkgConfig REQUIRED)
+if (NOT CYGWIN)
+  find_package(PkgConfig REQUIRED)
+endif()
 
 find_package(ASSIMP QUIET)
-if (NOT ASSIMP_FOUND)
+
+if (NOT ASSIMP_FOUND AND NOT CYGWIN)
   pkg_check_modules(ASSIMP assimp)
 endif()
+
 if (ASSIMP_FOUND)
   if( NOT ${ASSIMP_VERSION} VERSION_LESS "2.0.1150" )
     add_definitions(-DASSIMP_UNIFIED_HEADER_NAMES)
@@ -42,14 +46,22 @@
   set(ASSIMP_INCLUDE_DIRS)
 endif()
 
-pkg_check_modules(OGRE_OV OGRE OGRE-Overlay)
+if (CYGWIN)
+set(OGRE_OV_INCLUDE_DIRS "/usr/local/include/OGRE")
+set(OGRE_OV_LIBRARY_DIRS "/usr/local/lib")
+set(OGRE_OV_LIBRARIES "OgreMain")
+endif()
+
+if (NOT CYGWIN)
+  pkg_check_modules(OGRE_OV OGRE OGRE-Overlay)
+endif()
 
 # Old versions of OGRE (pre 1.9) included OGRE-Overlay in the main package
 # (i.e. there was no OGRE-Overlay component).  So if the above
 # pkg_check_modules() failed, try looking for just OGRE.
-if(NOT OGRE_OV_FOUND)
-  pkg_check_modules(OGRE_OV REQUIRED OGRE)
-endif(NOT OGRE_OV_FOUND)
+if(NOT OGRE_OV_FOUND AND NOT CYGWIN)
+  pkg_check_modules(OGRE_OV REQUIRED OGRE AND NOT CYGWIN)
+endif(NOT OGRE_OV_FOUND AND NOT CYGWIN)
 
 ## Find OGRE Plugin path (not necessarily platform-independent, I guess)
 execute_process(COMMAND
@@ -84,6 +96,7 @@
         CACHE FILEPATH "Pathname of library ${_lib}")
 endif(NOT DEFINED OGRE_OV_LIBRARIES_ABS)
 
+
 if(APPLE)
   FIND_LIBRARY(Cocoa_LIBRARIES Cocoa)
   set(rviz_ADDITIONAL_LIBRARIES ${Cocoa_LIBRARIES})
diff -urN src.orig/rviz/plugin_description.xml src.fix/rviz/plugin_description.xml
--- src.orig/rviz/plugin_description.xml	2015-10-13 10:52:04.000000000 -0700
+++ src.fix/rviz/plugin_description.xml	2015-11-19 09:12:17.909810700 -0800
@@ -1,4 +1,4 @@
-<library path="libdefault_plugin">
+<library path="cygdefault_plugin">
   <class name="rviz/Axes" type="rviz::AxesDisplay" base_class_type="rviz::Display">
     <description>
       Displays an axis at the Target Frame's origin.  &lt;a href="http://www.ros.org/wiki/rviz/DisplayTypes/Axes"&gt;More Information&lt;/a&gt;.
diff -urN src.orig/rviz/src/python_bindings/sip/CMakeLists.txt src.fix/rviz/src/python_bindings/sip/CMakeLists.txt
--- src.orig/rviz/src/python_bindings/sip/CMakeLists.txt	2015-10-13 10:52:04.000000000 -0700
+++ src.fix/rviz/src/python_bindings/sip/CMakeLists.txt	2015-11-19 09:13:44.921511000 -0800
@@ -41,7 +41,7 @@
 
 # maintain context for different named target
 set(rviz_sip_INCLUDE_DIRS ${rviz_INCLUDE_DIRS} "${PROJECT_SOURCE_DIR}/src" ${catkin_INCLUDE_DIRS})
-set(rviz_sip_LIBRARIES ${rviz_LIBRARIES} ${PROJECT_NAME})
+set(rviz_sip_LIBRARIES ${rviz_LIBRARIES} ${PROJECT_NAME} boost_system)
 set(rviz_sip_LIBRARY_DIRS ${rviz_LIBRARY_DIRS} ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_LIB_DESTINATION})
 set(rviz_sip_LDFLAGS_OTHER ${rviz_LDFLAGS_OTHER} -Wl,-rpath,\\"${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_LIB_DESTINATION}\\")
 
diff -urN src.orig/rviz/src/rviz/CMakeLists.txt src.fix/rviz/src/rviz/CMakeLists.txt
--- src.orig/rviz/src/rviz/CMakeLists.txt	2015-10-13 10:52:04.000000000 -0700
+++ src.fix/rviz/src/rviz/CMakeLists.txt	2015-11-19 09:12:17.920811800 -0800
@@ -211,7 +211,8 @@
                       PROPERTIES OUTPUT_NAME ${PROJECT_NAME})
 
 install(TARGETS executable ${PROJECT_NAME}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  RUNTIME DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
 )
 install(DIRECTORY ./
diff -urN src.orig/rviz/src/rviz/default_plugin/CMakeLists.txt src.fix/rviz/src/rviz/default_plugin/CMakeLists.txt
--- src.orig/rviz/src/rviz/default_plugin/CMakeLists.txt	2015-10-13 10:52:04.000000000 -0700
+++ src.fix/rviz/src/rviz/default_plugin/CMakeLists.txt	2015-11-19 09:12:17.924812200 -0800
@@ -115,7 +115,9 @@
 
 install(TARGETS default_plugin
   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION})
+  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  RUNTIME DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  )
 
 #message("TODO: stop installing libdefault_plugin.so into share/rviz after pluginlib starts looking for it in lib.")
 #install(TARGETS default_plugin
diff -urN src.orig/rviz/src/rviz/mesh_loader.cpp src.fix/rviz/src/rviz/mesh_loader.cpp
--- src.orig/rviz/src/rviz/mesh_loader.cpp	2015-10-13 10:52:04.000000000 -0700
+++ src.fix/rviz/src/rviz/mesh_loader.cpp	2015-11-19 09:12:17.928812600 -0800
@@ -53,6 +53,7 @@
 
 #include <ros/assert.h>
 
+#undef ASSIMP_UNIFIED_HEADER_NAMES
 #if defined(ASSIMP_UNIFIED_HEADER_NAMES)
 #include <assimp/Importer.hpp>
 #include <assimp/scene.h>
diff -urN src.orig/rviz/src/rviz/ogre_helpers/render_system.cpp src.fix/rviz/src/rviz/ogre_helpers/render_system.cpp
--- src.orig/rviz/src/rviz/ogre_helpers/render_system.cpp	2015-10-13 10:52:04.000000000 -0700
+++ src.fix/rviz/src/rviz/ogre_helpers/render_system.cpp	2015-11-19 09:12:17.933813100 -0800
@@ -30,7 +30,7 @@
 // This is required for QT_MAC_USE_COCOA to be set
 #include <QtCore/qglobal.h>
 
-#ifndef Q_OS_MAC
+#if !defined(Q_OS_MAC) && !defined(Q_WS_WIN)
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 #include <GL/glx.h>
@@ -119,7 +119,7 @@
 
 void RenderSystem::setupDummyWindowId()
 {
-#ifdef Q_OS_MAC
+#if defined(Q_OS_MAC) || defined(Q_WS_WIN)
   dummy_window_id_ = 0;
 #else
   Display *display = XOpenDisplay(0);
@@ -144,10 +144,14 @@
 
 void RenderSystem::loadOgrePlugins()
 {
+#ifdef __CYGWIN__
+  std::string plugin_prefix = "/usr/local/bin/";
+#else
   std::string plugin_prefix = get_ogre_plugin_path() + "/";
 #ifdef Q_OS_MAC
   plugin_prefix += "lib";
 #endif
+#endif
   ogre_root_->loadPlugin( plugin_prefix + "RenderSystem_GL" );
   ogre_root_->loadPlugin( plugin_prefix + "Plugin_OctreeSceneManager" );
   ogre_root_->loadPlugin( plugin_prefix + "Plugin_ParticleFX" );
@@ -341,7 +345,7 @@
   std::stringstream window_handle_stream;
   window_handle_stream << window_id;
 
-#ifdef Q_OS_MAC
+#if defined(Q_OS_MAC) || defined(__CYGWIN__)
   params["externalWindowHandle"] = window_handle_stream.str();
 #else
   params["parentWindowHandle"] = window_handle_stream.str();
diff -urN src.orig/rviz/src/rviz/ogre_helpers/render_widget.cpp src.fix/rviz/src/rviz/ogre_helpers/render_widget.cpp
--- src.orig/rviz/src/rviz/ogre_helpers/render_widget.cpp	2015-10-13 10:52:04.000000000 -0700
+++ src.fix/rviz/src/rviz/ogre_helpers/render_widget.cpp	2015-11-19 09:12:17.938813600 -0800
@@ -70,7 +70,7 @@
 #ifdef Q_OS_MAC
   uintptr_t win_id = winId();
 #else
-  unsigned int win_id = renderFrame->winId();
+  unsigned int win_id = (unsigned)(UINT_PTR)renderFrame->winId();
 #endif  
   QApplication::flush();
   QApplication::syncX();
diff -urN src.orig/rviz/src/rviz/yaml_config_reader.cpp src.fix/rviz/src/rviz/yaml_config_reader.cpp
--- src.orig/rviz/src/rviz/yaml_config_reader.cpp	2015-10-13 10:52:04.000000000 -0700
+++ src.fix/rviz/src/rviz/yaml_config_reader.cpp	2015-11-19 09:12:17.942814000 -0800
@@ -30,6 +30,8 @@
 #include <fstream>
 #include <sstream>
 
+#undef RVIZ_HAVE_YAMLCPP_05
+
 #ifdef RVIZ_HAVE_YAMLCPP_05
 #include <yaml-cpp/yaml.h>
 #else
